test/integration/fetch-collection/index.js

```diff
@@ -1,117 +1,138 @@
-              });
-            });
-
-            describe("Resource Objects/Primary Data", () => {
-              // "A logical collection of resources MUST be represented as
-              //  an array, even if it only contains one item or is empty."
-              it("should be an array under data", () => {
-                expect(res.body.data).to.be.an("array");
-              });
-
-              // "Unless otherwise noted, objects defined by this
-              //  specification MUST NOT contain any additional members."
-              it("should not contain extra members", () => {
-                const isAllowedKey = (key) =>
-                  ["type", "id", "attributes", "relationships", "links", "meta"].indexOf(key) !== -1;
-
-                if(!Object.keys(res.body.data[0]).every(isAllowedKey)) {
-                  throw new Error("Invalid Key!");
-                }
-              });
-
-              it("should contain links under each relationship", () => {
-                let liaisonRelationships = res.body.data.map(it => it.relationships.liaisons);
-                liaisonRelationships.forEach((it) => {
-                  // Every liaison should have a links object,
-                  // and self and related must not be outside of links.
-                  expect(it.links).to.be.an("object");
-                  expect(it.self).to.be.undefined;
-                  expect(it.related).to.be.undefined;
-
-                  expect(it.links.self).to.be.a("string");
-                  expect(it.data).to.not.be.undefined; //can be null, though
-                });
-              });
-            });
-          });
-        });
-      }).done();
+describe("Fetching Collection", () => {
+  let res, Agent;
 
-    Agent.request("GET", "/people?sort=gender")
-      .accept("application/vnd.api+json")
-      .promise()
-      .then((res) => {
-        describe("Fetching Ascending Gendered Collection", () => {
-          it("should have Jane above John", () => {
-            let johnJaneList = res.body.data.map((it) => it.attributes.name).filter((it) => {
-              return ["John Smith", "Jane Doe"].indexOf(it) > -1
-            });
-            expect(johnJaneList[0]).to.equal("Jane Doe");
-            expect(johnJaneList[1]).to.equal("John Smith");
-          });
-        })
-      }).done();
-
-    Agent.request("GET", "/people?sort=-name")
-      .accept("application/vnd.api+json")
-      .promise()
-      .then((res) => {
-        describe("Fetching Descended Sorted Name Collection", () => {
-          it("Should have John above Jane", () => {
-            let johnJaneList = res.body.data.map((it) => it.attributes.name).filter((it) => {
-              return ["John", "Jane"].indexOf(it.substring(0, 4)) > -1
-            });
-            expect(johnJaneList[0]).to.equal("John Smith");
-            expect(johnJaneList[1]).to.equal("Jane Doe");
-          });
+  before(done => {
```

Comment:
Hot damn this refactor is nice. Great call on using before() to simplify things.